signature-crate/src/digest/digestable.rs

```diff
@@ -0,0 +1,14 @@
+use crate::{digest::Digest, signature::Signature};
+
+/// Marker trait for `Signature` types computable as `S(H(m))` where:
+///
+/// - `S`: signature algorithm
+/// - `H`: hash (a.k.a. digest) function
+/// - `m`: message
+///
+/// For signature types that implement this trait, a blanket impl of
+/// `Signer` will be provided for all types that `impl digest::Signer`.
+pub trait Digestable: Signature {
+    /// Preferred `Digest` algorithm to use when computing this signature type.
+    type Digest: Digest;
+}
```

Comment:
Random thought: this pattern seems generally useful... perhaps it belongs in the `digest` crate?

Reply:
Hm, maybe, but I think we first should see one-two similar use-cases first, before including it into `digest`.

BTW is it me or `Digestable` feels reverse in a sense? It's like "we can digest this signature", while it should be "this signature is calculated from message digest". How about `DigestSignature`? It still feels a bit off, but I don't have better ideas right now.

Reply:
See 0ed928c which used `signature::digest::Signature` 😉